src/Mvc/Mvc.ViewFeatures/src/ViewComponentResult.cs

```diff
@@ -64,7 +64,15 @@ public override Task ExecuteResultAsync(ActionContext context)
             }
 
             var services = context.HttpContext.RequestServices;
-            var executor = services.GetRequiredService<IActionResultExecutor<ViewComponentResult>>();
+            var executor = services.GetService<IActionResultExecutor<ViewComponentResult>>();
```

Comment:
So one of the things that I added is some guardrails around things like returning `View()` from a controller without the services. I figure this is is the most likely way for someone to experience this particular problem.

I couldn't think of other things that need to be fixed, there aren't many APIs that the user calls that wouldn't work, it's mainly the view result stuff.

Additionally, Identity UI already registers what it needs in services, so we're good there.

Reply:
We could always add more if we get user reports for `GetRequiredService` failing.